tests/test_daq_reader.py

```diff
@@ -0,0 +1,161 @@
+                'raw_records_aqmon',
+                ]
+    dummy_version = strax.Config(
+        default=None,
+        track=True,
+        help="Extra option to make sure that we are getting a different lineage if we want to",
+    )
+    data_kind = dict(zip(provides, provides))
+    rechunk_on_save = False
+
+    def _path(self, chunk_i):
+        path = super()._path(chunk_i)
+        path_exists = os.path.exists(path)
+        print(f'looked for {chunk_i} on {path}. Is found = {path_exists}')
+        return path
+
+
+class TestDAQReader(unittest.TestCase):
+    """
+    Test DAQReader with a few chunks of amstrax data:
+    https://github.com/XAMS-nikhef/amstrax
+    """
+
+    run_id = '999999'
+    run_doc_name = 'rundoc_999999.json'
+    live_data_path = f'./live_data/{run_id}'
+    rundoc_file = 'https://raw.githubusercontent.com/XAMS-nikhef/amstrax_files/73681f112d748f6cd0e95045970dd29c44e983b0/data/rundoc_999999.json'  # noqa
+    data_file = 'https://raw.githubusercontent.com/XAMS-nikhef/amstrax_files/73681f112d748f6cd0e95045970dd29c44e983b0/data/999999.tar'  # noqa
+
+    # Tests
+    # -----
+    def test_make(self) -> None:
+        """
+        Test if we can run the daq-reader without chrashing and if we
+        actually stored the data after making it.
+        """
+        run_id = self.run_id
+        for target, plugin_class in self.st._plugin_class_registry.items():
+            self.st.make(run_id, target)
+            sleep(0.5)  # allow os to rename the file
+            if plugin_class.save_when >= strax.SaveWhen.TARGET:
+                self.assertTrue(
+                    self.st.is_stored(run_id, target),
+                )
+
+    def test_insert_deadtime(self):
+        """
+        In the DAQ reader, we need a mimimium quiet period to say where
+        we can start/end a chunk. Test that this information gets
+        propagated to the ARTIFICIAL_DEADTIME_CHANNEL (in
+        raw-records-aqmon) if we set this value to an unrealistic value
+        of 0.5 s.
+        """
+        st = self.st.new_context()
+        st.set_config({'safe_break_in_pulses': int(0.5e9),
+                       'dummy_version': 'test_insert_deadtime',
+                       })
+
+        with self.assertWarns(ArtificialDeadtimeInserted):
+            st.make(self.run_id, 'raw_records')
+
+        rr_aqmon = st.get_array(self.run_id, 'raw_records_aqmon')
+        self.assertTrue(len(rr_aqmon))
+
+    def test_invalid_setting(self):
+        """The safe break in pulses cannot be longer than the chunk size"""
+        st = self.st.new_context()
+        st.set_config({'safe_break_in_pulses': int(3600e9),
+                       'dummy_version': 'test_invalid_setting',
+                       })
+        with self.assertRaises(ValueError):
+            st.make(self.run_id, 'raw_records')
+
+    # Setup of tests like getting data and removing it after the test
+    # ---------------------------------------------------------------

```

Comment:
Hmm, I think this is not needed.

Reply:
sure can delete, just wanted to illustrate that there are three separate stages to this test:
 - testing
 - setups
 - tools

Reply:
👍 I've removed this over-verbosity to the docstring